Fix #765: Full Highfi: HelpActivity

SayantanBanerjee16 opened the pull request:
## Explanation

Fixes #765 

Ensured Help activity, both portrait, and landscape matches with the mocks. 
Created the landscape layout of help-item as both layouts have some difference.
Accessibility Scanner test passed.
## Mocks

Portrait mock :

https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/8e1bbf3c-3287-4149-b599-5d5a5486809d/HL-Help-Menu-

Landscape mock:

https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/a949ea57-5166-4081-bec8-ed4c16d319d0/HL-Help-Menu-

## ScreenShots

Previous portrait mode:

![portrait-previous](https://user-images.githubusercontent.com/43978143/76760133-1c4f7e00-67b3-11ea-9676-b9f33efae1ab.jpg)

Current portrait mode:

![portrait-current](https://user-images.githubusercontent.com/43978143/76760162-270a1300-67b3-11ea-8f12-3888e7f9dca6.jpg)

Previous Landscape mode:

![landscape - previous](https://user-images.githubusercontent.com/43978143/76760185-312c1180-67b3-11ea-93fb-2cc3abbb6be8.jpg)

Current Landscape mode:

![landscape-current](https://user-images.githubusercontent.com/43978143/76760203-3b4e1000-67b3-11ea-8e55-527285934069.jpg)

## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

SayantanBanerjee16 committed:
Help Activity made according to the mocks in both portrait and landscape

SayantanBanerjee16 committed:
.idea files reverted

SayantanBanerjee16 committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into hifi-helpactivity

SayantanBanerjee16 commented:
@veena14cs @nikitamarysolomanpvt

PTAL. Thanks :)

Accessibility Scanner test passed.

Portrait mock :


https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/8e1bbf3c-3287-4149-b599-5d5a5486809d/HL-Help-Menu-

Landscape mock:

https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/a949ea57-5166-4081-bec8-ed4c16d319d0/HL-Help-Menu-

Previous portrait mode:

![portrait-previous](https://user-images.githubusercontent.com/43978143/76760133-1c4f7e00-67b3-11ea-9676-b9f33efae1ab.jpg)

Current portrait mode:

![portrait-current](https://user-images.githubusercontent.com/43978143/76760162-270a1300-67b3-11ea-8f12-3888e7f9dca6.jpg)

Previous Landscape mode:

![landscape - previous](https://user-images.githubusercontent.com/43978143/76760185-312c1180-67b3-11ea-93fb-2cc3abbb6be8.jpg)

Current Landscape mode:

![landscape-current](https://user-images.githubusercontent.com/43978143/76760203-3b4e1000-67b3-11ea-8e55-527285934069.jpg)

veena14cs commented on app/src/main/res/layout-land/help_item.xml:
```diff
@@ -0,0 +1,32 @@
+    android:layout_height="wrap_content">
+
+    <TextView
+      android:id="@+id/recycler_item_text_view"
```
Please change this id to what this textview holds, meaning Ids must be contextual.

rt4914 replied:
Make sure that this is changed in test cases also, if it is getting used.

SayantanBanerjee16 replied:
Okay I will do this change once I reach home, am currently traveling.

SayantanBanerjee16 replied:
Done :)

veena14cs commented:
@mschanteltc can you please provide specs for landscape toolbar. Look at this landscape screenshot for help activity https://user-images.githubusercontent.com/43978143/76760203-3b4e1000-67b3-11ea-8e55-527285934069.jpg. Toolbar looks smaller.

nikitamarysolomanpvt commented on app/src/main/res/layout/help_activity.xml:
```diff
@@ -17,7 +17,8 @@
       android:id="@+id/help_fragment_placeholder"
       android:layout_width="match_parent"
-      android:layout_height="match_parent" />
+      android:layout_height="match_parent"
```
From  help_fragment please remove unused name space.

SayantanBanerjee16 replied:
Done :)

nikitamarysolomanpvt commented on app/src/main/res/layout/help_item.xml:
```diff
@@ -17,10 +17,12 @@
+      android:paddingBottom="20dp"
       android:text="@{viewmodel.title}"
       android:textColor="@color/oppiaPrimaryText"
       android:textSize="20sp"
```
In mock it seems 16sp. PTAL

SayantanBanerjee16 replied:
Updated :)

nikitamarysolomanpvt reviewed (commented):
Please add divider to the list.. as per mocks
<img width="370" alt="small" src="https://user-images.githubusercontent.com/54615666/76831374-38076280-684d-11ea-9227-281b5ba99699.png">

SayantanBanerjee16 commented:
> @mschanteltc can you please provide specs for landscape toolbar. Look at this landscape screenshot for help activity https://user-images.githubusercontent.com/43978143/76760203-3b4e1000-67b3-11ea-8e55-527285934069.jpg. Toolbar looks smaller.

Yes in all the screens designed in landscape so far in the App, the toolbar seems to be smaller than the mock. So Can you guide me where I should be looking at for making it like the mocks?

mschanteltc commented:
> @mschanteltc can you please provide specs for landscape toolbar. Look at this landscape screenshot for help activity https://user-images.githubusercontent.com/43978143/76760203-3b4e1000-67b3-11ea-8e55-527285934069.jpg. Toolbar looks smaller.

The landscape toolbar is the same height as portrait's. You can view the specs [here](https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/a949ea57-5166-4081-bec8-ed4c16d319d0/HL-Help-Menu-).

SayantanBanerjee16 committed:
Layout modification

SayantanBanerjee16 commented:
> Please add divider to the list.. as per mocks
> <img alt="small" width="370" src="https://user-images.githubusercontent.com/54615666/76831374-38076280-684d-11ea-9227-281b5ba99699.png">

Done :)

SayantanBanerjee16 committed:
Toolbar landscape mode created and changes done

SayantanBanerjee16 committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into hifi-helpactivity

SayantanBanerjee16 commented:
@rt4914 @veena14cs @nikitamarysolomanpvt

Done with all the NIT changes and also Toolbar landscape mode created so as to implement the toolbar as per the mocks. The divider also set.

Accessibility Scanner tests passed.

PTAL. Thanks :)

Current Portrait mode:-

![Screenshot_20200318-092442_Oppia](https://user-images.githubusercontent.com/43978143/76924089-5166e800-68fb-11ea-9b39-3e7c39559655.jpg)

Current Landscape mode:-

![Screenshot_20200318-092450_Oppia](https://user-images.githubusercontent.com/43978143/76924118-6774a880-68fb-11ea-904f-6a2400d7131c.jpg)

veena14cs commented:
There is shadow below toolbar. Please add shadow to the toolbar you can refer to other layouts for example  recently_played_fragment.

veena14cs commented:
> > @mschanteltc can you please provide specs for landscape toolbar. Look at this landscape screenshot for help activity https://user-images.githubusercontent.com/43978143/76760203-3b4e1000-67b3-11ea-8e55-527285934069.jpg. Toolbar looks smaller.
> 
> The landscape toolbar is the same height as portrait's. You can view the specs [here](https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/a949ea57-5166-4081-bec8-ed4c16d319d0/HL-Help-Menu-).

Thanks.

SayantanBanerjee16 committed:
Toolbar shadow implemented

SayantanBanerjee16 commented:
> There is shadow below toolbar. Please add shadow to the toolbar you can refer to other layouts for example recently_played_fragment.

@veena14cs Shadow implemented in the toolbar. PTAL. Thanks :)

Updated Portrait mode:

![portrait_updated](https://user-images.githubusercontent.com/43978143/76959915-86e1f480-6940-11ea-8c24-117a1f2708d8.jpg)

Updated Landscape mode:

![landscape_updated](https://user-images.githubusercontent.com/43978143/76959938-919c8980-6940-11ea-8e57-dab5049fd434.jpg)

nikitamarysolomanpvt commented:
> @rt4914 @veena14cs @nikitamarysolomanpvt
> 
> Done with all the NIT changes and also Toolbar landscape mode created so as to implement the toolbar as per the mocks. The divider also set.
> 
> Accessibility Scanner tests passed.
> 
> PTAL. Thanks :)
> 
> Current Portrait mode:-
> 
> ![Screenshot_20200318-092442_Oppia](https://user-images.githubusercontent.com/43978143/76924089-5166e800-68fb-11ea-9b39-3e7c39559655.jpg)
> 
> Current Landscape mode:-
> 
> ![Screenshot_20200318-092450_Oppia](https://user-images.githubusercontent.com/43978143/76924118-6774a880-68fb-11ea-904f-6a2400d7131c.jpg)

seems still not matching with mock list item seems have lill shadow at bottom, seems divider color is quite darker.. @mschanteltc  PTAL

SayantanBanerjee16 commented:
> > @rt4914 @veena14cs @nikitamarysolomanpvt
> > Done with all the NIT changes and also Toolbar landscape mode created so as to implement the toolbar as per the mocks. The divider also set.
> > Accessibility Scanner tests passed.
> > PTAL. Thanks :)
> > Current Portrait mode:-
> > ![Screenshot_20200318-092442_Oppia](https://user-images.githubusercontent.com/43978143/76924089-5166e800-68fb-11ea-9b39-3e7c39559655.jpg)
> > Current Landscape mode:-
> > ![Screenshot_20200318-092450_Oppia](https://user-images.githubusercontent.com/43978143/76924118-6774a880-68fb-11ea-904f-6a2400d7131c.jpg)
> 
> seems still not matching with mock item seems have lill shadow at bottom, seems divider color is quite darker.. @mschanteltc PTAL

@nikitamarysolomanpvt Well, I have already put the toolbar shadow if you are referring 'shadow' to it (Check the comment https://github.com/oppia/oppia-android/pull/818#issuecomment-600589545). 

And the divider color is made similar to the background color with the height of 2dp. Can you tell me what should be that color?

mschanteltc commented:
> > @rt4914 @veena14cs @nikitamarysolomanpvt
> > Done with all the NIT changes and also Toolbar landscape mode created so as to implement the toolbar as per the mocks. The divider also set.
> > Accessibility Scanner tests passed.
> > PTAL. Thanks :)
> > Current Portrait mode:-
> > ![Screenshot_20200318-092442_Oppia](https://user-images.githubusercontent.com/43978143/76924089-5166e800-68fb-11ea-9b39-3e7c39559655.jpg)
> > Current Landscape mode:-
> > ![Screenshot_20200318-092450_Oppia](https://user-images.githubusercontent.com/43978143/76924118-6774a880-68fb-11ea-904f-6a2400d7131c.jpg)
> 
> seems still not matching with mock item seems have lill shadow at bottom, seems divider color is quite darker.. @mschanteltc PTAL

I'm a bit confused which mocks I should be looking at here. The shadow @SayantanBanerjee16 implemented looks correct, but the comment @nikitamarysolomanpvt made after doesn't show it. Is the background of "Frequently Asked Questions (FAQs)" pure white (#FFFFFF) or a different shade?

SayantanBanerjee16 commented:
@mschanteltc Please refer to my comment https://github.com/oppia/oppia-android/pull/818#issuecomment-600589545 it is the final one. @nikitamarysolomanpvt, unfortunately, commented on a previous comment without the toolbar shadow.

For the colors,

the Text is written in #333333 (oppiaPrimaryText)
the item color surrounding the text #B3FFFFFF (white_70)
the rest of the Greyish background is in #E5E5E5

And similar Greyish color also implemented in divider (#E5E5E5) of height 2dp.

Please comment on any of the changes I had to make.

veena14cs commented:
LGTM.

veena14cs reviewed (approved):
LGTM.

rt4914 commented:
Passing this to @nikitamarysolomanpvt for re-check of her suggestions.

nikitamarysolomanpvt commented:
> > > @rt4914 @veena14cs @nikitamarysolomanpvt
> > > Done with all the NIT changes and also Toolbar landscape mode created so as to implement the toolbar as per the mocks. The divider also set.
> > > Accessibility Scanner tests passed.
> > > PTAL. Thanks :)
> > > Current Portrait mode:-
> > > ![Screenshot_20200318-092442_Oppia](https://user-images.githubusercontent.com/43978143/76924089-5166e800-68fb-11ea-9b39-3e7c39559655.jpg)
> > > Current Landscape mode:-
> > > ![Screenshot_20200318-092450_Oppia](https://user-images.githubusercontent.com/43978143/76924118-6774a880-68fb-11ea-904f-6a2400d7131c.jpg)
> > 
> > 
> > seems still not matching with mock item seems have lill shadow at bottom, seems divider color is quite darker.. @mschanteltc PTAL
> 
> I'm a bit confused which mocks I should be looking at here. The shadow @SayantanBanerjee16 implemented looks correct, but the comment @nikitamarysolomanpvt made after doesn't show it. Is the background of "Frequently Asked Questions (FAQs)" pure white (#FFFFFF) or a different shade?

@mschanteltc I was mentioning the screenshots are not matching with mock.. 
https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/8e1bbf3c-3287-4149-b599-5d5a5486809d/HL-Help-Menu-/
In mock list item seems to have a shadow/elivation or a light border. Also divider color seems darker... PTAL
<img width="392" alt="small" src="https://user-images.githubusercontent.com/54615666/77037703-e9d29a80-69d7-11ea-8f72-292aae6c76a3.png">

SayantanBanerjee16 committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into hifi-helpactivity

mschanteltc commented:
> > > > @rt4914 @veena14cs @nikitamarysolomanpvt
> > > > Done with all the NIT changes and also Toolbar landscape mode created so as to implement the toolbar as per the mocks. The divider also set.
> > > > Accessibility Scanner tests passed.
> > > > PTAL. Thanks :)
> > > > Current Portrait mode:-
> > > > ![Screenshot_20200318-092442_Oppia](https://user-images.githubusercontent.com/43978143/76924089-5166e800-68fb-11ea-9b39-3e7c39559655.jpg)
> > > > Current Landscape mode:-
> > > > ![Screenshot_20200318-092450_Oppia](https://user-images.githubusercontent.com/43978143/76924118-6774a880-68fb-11ea-904f-6a2400d7131c.jpg)
> > > 
> > > 
> > > seems still not matching with mock item seems have lill shadow at bottom, seems divider color is quite darker.. @mschanteltc PTAL
> > 
> > 
> > I'm a bit confused which mocks I should be looking at here. The shadow @SayantanBanerjee16 implemented looks correct, but the comment @nikitamarysolomanpvt made after doesn't show it. Is the background of "Frequently Asked Questions (FAQs)" pure white (#FFFFFF) or a different shade?
> 
> @mschanteltc I was mentioning the screenshots are not matching with mock..
> https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/8e1bbf3c-3287-4149-b599-5d5a5486809d/HL-Help-Menu-/
> In mock list item seems to have a shadow/elivation or a light border. Also divider color seems darker... PTAL
> <img alt="small" width="392" src="https://user-images.githubusercontent.com/54615666/77037703-e9d29a80-69d7-11ea-8f72-292aae6c76a3.png">

The separation line used in the mock is #000000 with a 12% opacity, however the one in the screenshot looks fine as well. What color does it use? If we can get the background of the list to be #FFFFFF at 100% opacity, the contrast can make the separation line stand out more

SayantanBanerjee16 commented:
> > > > > @rt4914 @veena14cs @nikitamarysolomanpvt
> > > > > Done with all the NIT changes and also Toolbar landscape mode created so as to implement the toolbar as per the mocks. The divider also set.
> > > > > Accessibility Scanner tests passed.
> > > > > PTAL. Thanks :)
> > > > > Current Portrait mode:-
> > > > > ![Screenshot_20200318-092442_Oppia](https://user-images.githubusercontent.com/43978143/76924089-5166e800-68fb-11ea-9b39-3e7c39559655.jpg)
> > > > > Current Landscape mode:-
> > > > > ![Screenshot_20200318-092450_Oppia](https://user-images.githubusercontent.com/43978143/76924118-6774a880-68fb-11ea-904f-6a2400d7131c.jpg)
> > > > 
> > > > 
> > > > seems still not matching with mock item seems have lill shadow at bottom, seems divider color is quite darker.. @mschanteltc PTAL
> > > 
> > > 
> > > I'm a bit confused which mocks I should be looking at here. The shadow @SayantanBanerjee16 implemented looks correct, but the comment @nikitamarysolomanpvt made after doesn't show it. Is the background of "Frequently Asked Questions (FAQs)" pure white (#FFFFFF) or a different shade?
> > 
> > 
> > @mschanteltc I was mentioning the screenshots are not matching with mock..
> > https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/8e1bbf3c-3287-4149-b599-5d5a5486809d/HL-Help-Menu-/
> > In mock list item seems to have a shadow/elivation or a light border. Also divider color seems darker... PTAL
> > <img alt="small" width="392" src="https://user-images.githubusercontent.com/54615666/77037703-e9d29a80-69d7-11ea-8f72-292aae6c76a3.png">
> 
> The separation line used in the mock is #000000 with a 12% opacity, however the one in the screenshot looks fine as well. What color does it use? If we can get the background of the list to be #FFFFFF at 100% opacity, the contrast can make the separation line stand out more

@mschanteltc For the colors,

the Text is written in #333333 (oppiaPrimaryText)
the item color surrounding the text #B3FFFFFF (white_70)
the rest of the Greyish background is in #E5E5E5

And similar Greyish color also implemented in divider (#E5E5E5) of height 2dp.

So finally what should be each of these colors?
Like you mentioned about #FFFFFF, is it about item background or whole list background?

mschanteltc commented:
> > > > > > @rt4914 @veena14cs @nikitamarysolomanpvt
> > > > > > Done with all the NIT changes and also Toolbar landscape mode created so as to implement the toolbar as per the mocks. The divider also set.
> > > > > > Accessibility Scanner tests passed.
> > > > > > PTAL. Thanks :)
> > > > > > Current Portrait mode:-
> > > > > > ![Screenshot_20200318-092442_Oppia](https://user-images.githubusercontent.com/43978143/76924089-5166e800-68fb-11ea-9b39-3e7c39559655.jpg)
> > > > > > Current Landscape mode:-
> > > > > > ![Screenshot_20200318-092450_Oppia](https://user-images.githubusercontent.com/43978143/76924118-6774a880-68fb-11ea-904f-6a2400d7131c.jpg)
> > > > > 
> > > > > 
> > > > > seems still not matching with mock item seems have lill shadow at bottom, seems divider color is quite darker.. @mschanteltc PTAL
> > > > 
> > > > 
> > > > I'm a bit confused which mocks I should be looking at here. The shadow @SayantanBanerjee16 implemented looks correct, but the comment @nikitamarysolomanpvt made after doesn't show it. Is the background of "Frequently Asked Questions (FAQs)" pure white (#FFFFFF) or a different shade?
> > > 
> > > 
> > > @mschanteltc I was mentioning the screenshots are not matching with mock..
> > > https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/8e1bbf3c-3287-4149-b599-5d5a5486809d/HL-Help-Menu-/
> > > In mock list item seems to have a shadow/elivation or a light border. Also divider color seems darker... PTAL
> > > <img alt="small" width="392" src="https://user-images.githubusercontent.com/54615666/77037703-e9d29a80-69d7-11ea-8f72-292aae6c76a3.png">
> > 
> > 
> > The separation line used in the mock is #000000 with a 12% opacity, however the one in the screenshot looks fine as well. What color does it use? If we can get the background of the list to be #FFFFFF at 100% opacity, the contrast can make the separation line stand out more
> 
> @mschanteltc For the colors,
> 
> the Text is written in #333333 (oppiaPrimaryText)
> the item color surrounding the text #B3FFFFFF (white_70)
> the rest of the Greyish background is in #E5E5E5
> 
> And similar Greyish color also implemented in divider (#E5E5E5) of height 2dp.
> 
> So finally what should be each of these colors?
> Like you mentioned about #FFFFFF, is it about item background or whole list background?

Only change "the item color surrounding the text" from #B3FFFFFF (white_70) to  #FFFFFFFF (white_100)

SayantanBanerjee16 committed:
color update

SayantanBanerjee16 committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into hifi-helpactivity

SayantanBanerjee16 commented:
@mschanteltc @nikitamarysolomanpvt Done with the color change. PTAL if it is good to go or there are some additional changes to it. Thanks :)

Current Portrait mode:

![Screenshot_20200322-215148_Oppia](https://user-images.githubusercontent.com/43978143/77255203-c9cb0180-6c8b-11ea-8183-180769f02acc.jpg)

Current Landscape mode:

![Screenshot_20200322-215156_Oppia](https://user-images.githubusercontent.com/43978143/77255212-d64f5a00-6c8b-11ea-8afa-714ba54e733d.jpg)

mschanteltc commented:
> @mschanteltc @nikitamarysolomanpvt Done with the color change. PTAL if it is good to go or there are some additional changes to it. Thanks :)
> 
> Current Portrait mode:
> 
> ![Screenshot_20200322-215148_Oppia](https://user-images.githubusercontent.com/43978143/77255203-c9cb0180-6c8b-11ea-8183-180769f02acc.jpg)
> 
> Current Landscape mode:
> 
> ![Screenshot_20200322-215156_Oppia](https://user-images.githubusercontent.com/43978143/77255212-d64f5a00-6c8b-11ea-8afa-714ba54e733d.jpg)

LGTM, thank you!

nikitamarysolomanpvt commented:
@SayantanBanerjee16 PTAL on the screenshot I am adding the dummy item in the list so that we may check the list divider colour and list item border well. Please add screenshots with more than one list item so that @mschanteltc can guide you. @mschanteltc PTAL on the below screenshot. I feel the design is still not matching. Either it can be because of divider colour or the list item shadow/border.
<img width="946" alt="small" src="https://user-images.githubusercontent.com/54615666/77275119-86997e80-6cdd-11ea-9811-3e3a9611f6c5.png">

mschanteltc commented:
> @SayantanBanerjee16 PTAL on the screenshot I am adding the dummy item in the list so that we may check the list divider colour and list item border well. Please add screenshots with more than one list item so that @mschanteltc can guide you. @mschanteltc PTAL on the below screenshot. I feel the design is still not matching. Either it can be because of divider colour or the list item shadow/border.
> <img alt="small" width="946" src="https://user-images.githubusercontent.com/54615666/77275119-86997e80-6cdd-11ea-9811-3e3a9611f6c5.png">

I updated the mock once more so that the shadow is removed. The border I used is #000000 at 12% opacity. Once these are done, the mocks should have the same settings as what is being implemented.

SayantanBanerjee16 committed:
divider color update

SayantanBanerjee16 commented:
@nikitamarysolomanpvt @mschanteltc 
I have updated the divider color (#000000 at 12% opacity).  PTAL.

Also is there any left and right side border I need to implement?

Current portrait mode with 3 items(dummy items not pushed in code):-

![Screenshot_20200323-094917_Oppia](https://user-images.githubusercontent.com/43978143/77281365-d92e6700-6cec-11ea-9437-b9401c66e102.jpg)

Current landscape mode with 3 items(dummy items not pushed in code):-

![Screenshot_20200323-094923_Oppia](https://user-images.githubusercontent.com/43978143/77281392-e9dedd00-6cec-11ea-8d0f-9dd21e65906e.jpg)

nikitamarysolomanpvt commented on app/src/main/res/layout-land/help_item.xml:
```diff
@@ -0,0 +1,44 @@
+  </data>
+
+  <androidx.constraintlayout.widget.ConstraintLayout
+    android:layout_width="match_parent"
```
Please see the mock there is a border of 1dp given to the list item. Seems like you may revert your last commit and add this border and this would fix over all.
<img width="1117" alt="small" src="https://user-images.githubusercontent.com/54615666/77282694-a5edd700-6cf0-11ea-8f77-3ecb609acf98.png">

SayantanBanerjee16 committed:
border added

SayantanBanerjee16 committed:
borders color modified

SayantanBanerjee16 commented:
@nikitamarysolomanpvt Implemented the border as per the mocks. PTAL. Thanks :)

Portrait mode:
![Screenshot_20200323-161833_Oppia](https://user-images.githubusercontent.com/43978143/77309457-b61eaa00-6d22-11ea-8a8f-7a2b8cdcb25a.jpg)

Landscape mode:
![Screenshot_20200323-161841_Oppia](https://user-images.githubusercontent.com/43978143/77309495-c6cf2000-6d22-11ea-9bed-aeadfc360385.jpg)

nikitamarysolomanpvt commented:
@SayantanBanerjee16 this change looks almost as per mock, but still i think we can improve this by introducing a border drawable to the item.

```
<?xml version="1.0" encoding="utf-8"?>
<layout xmlns:android="http://schemas.android.com/apk/res/android"
  xmlns:app="http://schemas.android.com/apk/res-auto">

  <data class="HelpItemBinding">

    <variable
      name="viewModel"
      type="org.oppia.app.help.HelpViewModel" />
  </data>

  <TextView
    android:id="@+id/help_item_text_view"
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:background="@drawable/help_item_background_border"
    android:fontFamily="sans-serif"
    android:paddingStart="16dp"
    android:paddingTop="20dp"
    android:paddingEnd="16dp"
    android:paddingBottom="20dp"
    android:text="@{viewModel.title}"
    android:textColor="@color/oppiaPrimaryText"
    android:textSize="16sp"
    app:layout_constraintStart_toStartOf="parent"
    app:layout_constraintTop_toTopOf="parent" />
</layout>

help_item_background_border.xml
<?xml version="1.0" encoding="utf-8"?>
<shape xmlns:android="http://schemas.android.com/apk/res/android"
  android:shape="rectangle">
  <solid android:color="@color/white" />
  <stroke
    android:width="1dp"
    android:color="@color/black_12" />
</shape>


```

 Please find the screen shot of the resulting change compared with yours
With view as divider and without border -- screen shot of current PR
<img width="1002" alt="small" src="https://user-images.githubusercontent.com/54615666/77426384-61e6f900-6dfa-11ea-89a6-6064a77606ba.png">

With border of 1dp
<img width="1028" alt="small" src="https://user-images.githubusercontent.com/54615666/77426178-00bf2580-6dfa-11ea-8478-ae8eb6ae9b5e.png">

@rt4914 @mschanteltc PTAL any suggestions.

mschanteltc commented:
> @SayantanBanerjee16 this change looks almost as per mock, but still i think we can improve this by introducing a border drawable to the item.
> 
> ```
> <?xml version="1.0" encoding="utf-8"?>
> <layout xmlns:android="http://schemas.android.com/apk/res/android"
>   xmlns:app="http://schemas.android.com/apk/res-auto">
> 
>   <data class="HelpItemBinding">
> 
>     <variable
>       name="viewModel"
>       type="org.oppia.app.help.HelpViewModel" />
>   </data>
> 
>   <TextView
>     android:id="@+id/help_item_text_view"
>     android:layout_width="match_parent"
>     android:layout_height="wrap_content"
>     android:background="@drawable/help_item_background_border"
>     android:fontFamily="sans-serif"
>     android:paddingStart="16dp"
>     android:paddingTop="20dp"
>     android:paddingEnd="16dp"
>     android:paddingBottom="20dp"
>     android:text="@{viewModel.title}"
>     android:textColor="@color/oppiaPrimaryText"
>     android:textSize="16sp"
>     app:layout_constraintStart_toStartOf="parent"
>     app:layout_constraintTop_toTopOf="parent" />
> </layout>
> 
> help_item_background_border.xml
> <?xml version="1.0" encoding="utf-8"?>
> <shape xmlns:android="http://schemas.android.com/apk/res/android"
>   android:shape="rectangle">
>   <solid android:color="@color/white" />
>   <stroke
>     android:width="1dp"
>     android:color="@color/black_12" />
> </shape>
> ```
> 
> Please find the screen shot of the resulting change compared with yours
> With view as divider and without border -- screen shot of current PR
> <img alt="small" width="1002" src="https://user-images.githubusercontent.com/54615666/77426384-61e6f900-6dfa-11ea-89a6-6064a77606ba.png">
> 
> With border of 1dp
> <img alt="small" width="1028" src="https://user-images.githubusercontent.com/54615666/77426178-00bf2580-6dfa-11ea-8478-ae8eb6ae9b5e.png">
> 
> @rt4914 @mschanteltc PTAL any suggestions.

This change also works, LGTM!

SayantanBanerjee16 committed:
border done

SayantanBanerjee16 commented:
@nikitamarysolomanpvt Incorporated the border. PTAL. Thanks :)

Landscape:

![Screenshot_20200325-125447_Oppia](https://user-images.githubusercontent.com/43978143/77512241-a0cb8c00-6e98-11ea-84a5-40086babf58f.jpg)

Portrait:

![Screenshot_20200325-125456_Oppia](https://user-images.githubusercontent.com/43978143/77512474-1f282e00-6e99-11ea-8616-89ced7eda4ae.jpg)

SayantanBanerjee16 committed:
nit change

nikitamarysolomanpvt reviewed (approved):
LGTM

rt4914 commented:
@SayantanBanerjee16 Thanks a lot. For now i think we won't be able to merge this PR because of test cases failing on CirclCI. Please take latest code from `develop` and merge in this. If it passes the tests, I will merge it or else I will take care of it. Thanks.

SayantanBanerjee16 committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into hifi-helpactivity

SayantanBanerjee16 commented:
@rt4914 I pulled the changes from master and pushed but again CircleCI tests failed. What to do?

rt4914 commented:
> @rt4914 I pulled the changes from master and pushed but again CircleCI tests failed. What to do?

Okay in this case, I will checkout another branch from this PR and create a different PR. Thanks a lot I will take care of this from here.

SayantanBanerjee16 commented:
> > @rt4914 I pulled the changes from master and pushed but again CircleCI tests failed. What to do?
> 
> Okay in this case, I will checkout another branch from this PR and create a different PR. Thanks a lot I will take care of this from here.

Alright, thanks :)

SayantanBanerjee16 committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into hifi-helpactivity

SayantanBanerjee16 committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into hifi-helpactivity

SayantanBanerjee16 commented:
@rt4914 I have taken the pull containing circle_ci tests updation and pushed but still it failed to pass. Any help on what to do?

rt4914 commented:
@SayantanBanerjee16 The issue is that while you take the latest develop and push it and if there is some update in develop branch while the tests are running, they will fail. It only happens when two contributors are try to merge at the same time because once someone updates the develop your branch again is behind develop.

@SayantanBanerjee16 So can you just take one more merge within next 1hour? Because I think no one will merge in next 1 hour.

SayantanBanerjee16 committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into hifi-helpactivity

SayantanBanerjee16 commented:
@rt4914 Great it passed now. Please merge it. Thanks :)
